qutebrowser/misc/editor.py

```diff
@@ -121,21 +122,72 @@ def edit(self, text):
             return
 
         self._remove_file = True
-        self._start_editor()
+
+        # Here we calculate the line and column of the caret based on its
+        # position and the given text.
+        #
+        # NOTE: Both line and column are 1-based indexes, because that's what
+        # most editors use as line and column starting index.
+        # By "most" we mean at least vim, nvim, gvim, emacs, atom, sublimetext,
+        # notepad++, brackets, visual studio, QtCreator and so on.
+        #
+        # To find the line we just count how many newlines there are before
+        # the caret and add 1.
+        #
+        # To find the column we calculate the difference between the caret and
+        # the last newline before the caret.
+        #
+        # For example in the text `aaa\nbb|bbb` (| represents the caret):
+        # caret_position = 6
+        # text[:caret_position] = `aaa\nbb`
+        # text[:caret_psotion].count('\n') = 1
+        # caret_position - text[:caret_position].rfind('\n') = 3
+        #
+        # Thus line, column = 2, 3, and the caret is indeed in the second
+        # line, third column
+        line = text[:caret_position].count('\n') + 1
+        column = caret_position - text[:caret_position].rfind('\n')
```

Comment:
With the issues you've mentioned with getting the column, I feel like we should just drop and not support it. Probably nice enough to land on the right line already.

Reply:
I'm not sure about this, the calculation is correct, provided the column number represents unicode code-points. While the feature is broken for the vi family of editors I feel it will probably work correctly with editors such as atom or sublime text, but I have yet to test this.

In any case, I can remove it if you feel that's the best course of action.

Reply:
I think it depends whether other editors actually have a way to set the cursor column when invoking. The problem with having it but it not working correctly with vim is that I get to deal with the bug reports then :wink:

Do you know whether the cursor can be set like this in other editors?

Reply:
Thanks to Ape's suggestion we have a working command for vim and family.

I also tested atom and sublime text 3, both have no problems with multi-byte or zero-width characters, and both need the same command `'["atom", "{file}:{line}:{column}"]'`